scripts/migration/initialize_menas.py

```diff
@@ -0,0 +1,177 @@
+        prog='initialize_menas',
+        description='Menas MongoDB initialization script.',
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter  # prints default values, too, on help (-h)
+    )
+
+    parser.add_argument('-n', '--dryrun', action='store_true', default=DEFAULT_DRYRUN,
+                        help="if specified, skip the actual synchronization, just print what would be copied over.")
+    parser.add_argument('-v', '--verbose', action="store_true", default=DEFAULT_VERBOSE,
+                        help="prints extra information while running.")
+
+    parser.add_argument('target', metavar="TARGET", help="connection string for target MongoDB")
+
+    parser.add_argument('-t', '--target-database', dest="targetdb", default=DEFAULT_DB_NAME,
+                        help="Name of db on target to migrate to.")
+
+    return parser.parse_args()
+
+
+def initialize_menas_db(menas_db: MenasDb) -> None:
+    # create necessary collections
+    create_collections(menas_db.mongodb, MIGRATING_COLLECTIONS)
+
+    # create necessary indices
+    for collection in MIGRATING_COLLECTIONS:
+        indices = INDICES.get(collection, None)  # returns None if key is not found
+        if indices:
+            print(f"Initialization indices ({len(indices)}) on Menas collection '{collection}':")
+            initialize_collection_indices(menas_db.mongodb, collection, indices)
+        else:
+            print(f"Collection {collection} needs no index setup.")
+        print("")
+
+    # create db_version record - last as a sign of correctly initialized Menas DB
+    print("'db_version' initialization:")
+    create_db_version(menas_db.mongodb)
+
+
+def create_db_version(db: Database) -> None:
+    """
+    Attempts to set collection #DB_VERSION_COLLECTION with a record having version=1
+    """
+
+    collection = db[DB_VERSION_COLLECTION]
+    version_record = collection.find_one()
+    if verbose:
+        print(f"  Version record retrieval attempt: {version_record}")
+
+    # either version=1 record exists, or create new
+    if version_record:
+        if version_record["version"] != 1:
+            # failing on incompatible version, because that may corrupt data
+            raise MenasDbVersionError(f"Existing incompatible version record has been found in"
+                                      f" {DB_VERSION_COLLECTION}{self.hint}: {version_record}")
+        else:
+            print("  Existing db_version version=1 record found.")
+    else:
+        insert_result = collection.insert_one({"version": 1})
+        print(f"  Created version=1 record with id {insert_result.inserted_id}")
+    print("")
+
+
+def create_collections(db: Database, collection_names: List[str]):
+    print("Initialization of migration-related collections")
+    existing_collections = db.list_collection_names()
+    if len(existing_collections) != 0:
+        print(f"Warning, there are unexpected existing collections in the DB already: {existing_collections}")
```

Comment:
When I run init on a "bad" already existing DB. I get a warning that I am missing collection and then it goes on about indexes and then it fails. 

I need it to fail the second it sees that collections with our names are already in use

Reply:
```shell
Menas mongo initialization
Running with settings: dryrun=False, verbose=True
  target connection-string: mongodb://localhost:27017
  target DB: migration_aws_1

DB 'migration_aws_1' does not contain record version (DB migration_aws_1 (target db) does not contain collection db_version!)

Initializing DB 'migration_aws_1' for Menas...

Initialization of migration-related collections
Warning, there are unexpected existing collections in the DB already: ['dataset_v1', 'mapping_table_v1', 'schema_v1', 'attachment_v1']
  BEFORE: Aiming to create ['schema_v1', 'dataset_v1', 'mapping_table_v1', 'run_v1', 'attachment_v1'], found exising ['dataset_v1', 'mapping_table_v1', 'schema_v1', 'attachment_v1'].
  Creating missing collections ['run_v1'] ...
  AFTER: existing collections: ['dataset_v1', 'mapping_table_v1', 'schema_v1', 'run_v1', 'attachment_v1']

Initialization indices (1) on Menas collection 'schema_v1':
  BEFORE: index info: {'_id_': {'v': 2, 'key': [('_id', 1)]}}
  creating index for key [('name', 1), ('version', 1)], unique=True
  AFTER: index info: {'_id_': {'v': 2, 'key': [('_id', 1)]}, 'name_1_version_1': {'v': 2, 'unique': True, 'key': [('name', 1), ('version', 1)]}}

Initialization indices (1) on Menas collection 'dataset_v1':
  BEFORE: index info: {'_id_': {'v': 2, 'key': [('_id', 1)]}}
  creating index for key [('name', 1), ('version', 1)], unique=True
Traceback (most recent call last):
  File "/Users/.../Development/enceladus/scripts/migration/initialize_menas.py", line 141, in run
    target_db.check_db_version()
  File "/Users/.../Development/enceladus/scripts/migration/menas_db.py", line 84, in check_db_version
    raise MenasNoDbVersionRecordError(f"DB {self.mongodb.name}{self.hint} does not contain collection {DB_VERSION_COLLECTION}!")
menas_db.MenasNoDbVersionRecordError: DB migration_aws_1 (target db) does not contain collection db_version!

During handling of the above exception, another exception occurred:
```